### PR TITLE
Introduce an option to disable admin kubeconfig endpoints

### LINKED ISSUE #6245: Disabling non-oidc kubeconfig endpoints if oidc kubeconfig is enabled
### Description of the feature you would like to add / User story

It is currently possible to issue a long-term token admin kubeconfig via the KKP API by using your own short term bearer token.
You could issue a token with the following command:

```bash
curl -L -s ' https://dev.kubermatic.io/api/v2/projects/{project_id}/clusters/{cluster_id}/kubeconfig ' \
 -H 'accept: application/json, text/plain, */*' \
 -H 'authorization: Bearer {token}'
```

This endpoint is used to download an admin kubeconfig.

### Solution details

In terms of security, we should disable generating admin kubeconfig endpoint if the OIDC kubeconfig feature is enabled.

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**
**What this PR does / why we need it**:
This PR does the following things:
1. For API, disable admin kubeconfig endpoints when DisableAdminKubeconfig is enabled in KubermaticSettings
2. For the frontend, the option to disable admin kubeconfig feature from the admin panel.

**Which issue(s) this PR fixes**:

Fixes #6245

**What type of PR is this?**

/kind feature

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Support for disabling admin kubeconfig endpoint
```

**Documentation**:

```documentation
TBD
```

### CODE REVIEW

**Comment by ahmedwaleedmalik** [comment]
Requires https://github.com/kubermatic/dashboard/pull/6247 to continue

**Review by Waseem826 — APPROVED** [review]
/approve

File: modules/web/src/app/settings/admin/defaults/template.html [hunk 1]
```diff
@@ -212,13 +210,39 @@
           <km-spinner-with-confirmation [isSaved]="isEqual(settings.enableOIDCKubeconfig, apiSettings.enableOIDCKubeconfig)"></km-spinner-with-confirmation>
         </div>
 
+        <div fxLayout="row">
+          <div fxFlex="16%"
+               fxLayoutAlign=" center"
+               class="entry-label">
+            <span>Disable Admin Kubeconfig</span>
+            <div class="km-icon-info km-pointer"
+                 matTooltip="Disable the option to download admin kubeconfig and API endpoints."></div>
```
Comment (pkprzekwas) [inline modules/web/src/app/settings/admin/defaults/template.html:219]: nit: I'm a bit confused about wording here. How about?
```suggestion
                 matTooltip="Disable the option to download admin kubeconfig and corresponding API endpoint."></div>
```

Comment (ahmedwaleedmalik) [reply modules/web/src/app/settings/admin/defaults/template.html]: Looks better 🚀

**Comment by ahmedwaleedmalik** [comment]
/approve

Since it's already LGTM'd by the API and UI owners as well.

**Pull request merged**